### DISCUSSION TITLE
Notes for building Stride.Android.sln

### DISCUSSION THREAD

**Discussion by tagcode** [stride3d/stride#1601] [General]
I took some notes while getting Stride.Android.sln to build. These steps I took. This might help others. (2023 Feb 4th) 

# Android Studio
1. Install Android Studio
2. Install Android SDK Platforms (I ended up installing almost all of them)
3. Install NDK (Side by side). 
<img width="723" alt="AndroidStudio1" src="https://user-images.githubusercontent.com/5236113/216766833-abf69d85-fbea-41e0-81e6-60174f71ceaf.png">
<img width="731" alt="AndroidStudio2" src="https://user-images.githubusercontent.com/5236113/216766834-f8af9d37-a1f9-4ee5-8bc8-637f55d1bd03.png">

# Java
1. Install [JDK 11.0.17](https://www.oracle.com/java/technologies/javase/jdk11-archive-downloads.html). You must register to Oracle to be allowed to download it.

# Visual Studio
1. Install Xamarin
2. Install .NET SDK for Android, Android SDK Setup
<img width="960" alt="VisualStudio1" src="https://user-images.githubusercontent.com/5236113/216767023-cda17b1d-f399-4423-afd1-31843ad9b489.png">
<img width="960" alt="VisualStudio2" src="https://user-images.githubusercontent.com/5236113/216767025-ab7cff29-e534-4a89-baaf-e0fa716944e6.png">

# Environment Variables
1. JAVA_HOME = C:\Program Files\Java\jdk-11.0.17
2. JavaSdkDirectory = C:\Program Files\Java\jdk-11.0.17
3. AndroidSdkDirectory = C:\Users\username\AppData\Local\Android\Sdk
4. AndroidNdkDirectory = C:\Users\username\AppData\Local\Android\Sdk\ndk\25.2.9519653   **Check the path here**
5. StridePlatforms = Windows;Android
6. TRACEDESIGNTIME = true

# Stride.Android.sln
1. Clone from [github](https://github.com/stride3d/stride)
2. Edit: sources/native/Stide.Native.targets
3. Remove 32bit support: 'armeabi-v7a' and 'x86'
<img width="1639" alt="StrideAndroid" src="https://user-images.githubusercontent.com/5236113/216767195-487f8931-cd71-46c3-b626-0c25952ef134.png">
<img width="1637" alt="StrideAndroid2" src="https://user-images.githubusercontent.com/5236113/216767541-912eab2d-8832-4064-b94c-d7e1cdddf4ff.png">

# Sample Project
1. Create sample project from Stride Launcher. For example **GameMenu**. Choose Windows only. Then afterwards, in game studio, add Android support. 
2. Edit Game.Android.csproj
```
    <TargetFramework>net6.0-android33.0</TargetFramework>
    <SupportedOSPlatformVersion>25.0</SupportedOSPlatformVersion>
```
3. Edit also **RootNamespace**. Must be unique in the solution, e.g. "Game.Android".
```
<RootNamespace>GameMenu.Android</RootNamespace>
```
4. Edit Game.csproj
```
    <TargetFrameworks>net6.0-windows;net6.0-android33.0</TargetFrameworks>
```
5. Edit Configure startup projects >> Configuration Properties >> Deploy [x]
6. Edit on android project: Resources/values/strings.xml
```
<resources>
    <string name="app_name">Name</string>
    <string name="app_text">Text</string>
</resources>
```
7. View/Other Window/Package Management Console >> dotnet workload restore

<img width="970" alt="GameSolution1" src="https://user-images.githubusercontent.com/5236113/216767380-2b7a1920-124e-4afa-aabb-c07a6bc0b07f.png">
<img width="590" alt="GameSolution2" src="https://user-images.githubusercontent.com/5236113/216767383-cfc7e213-6e38-48f4-b0c1-61ed52828861.png">
<img width="506" alt="GameSolution3" src="https://user-images.githubusercontent.com/5236113/216767457-bedc9de7-e3fc-46fa-9844-08afa23b7791.png">
<img width="400" alt="GameSolution4" src="https://user-images.githubusercontent.com/5236113/216767460-376fa986-92b3-45ab-a9d7-23aae4cd96e3.png">

**Comment by tebjan** [comment]
Great, is this similar to the changes in this PR? https://github.com/stride3d/stride/pull/1456

**Reply by tagcode** [reply]
> Great, is this similar to the changes in this PR? #1456

Yes. I wrote the notes on the "Sample Project"-section with the release version of Stride. 

I can test later with the github Main version to check if it the fix works.

**Reply by tagcode** [reply]
Oh the PR wasn't merged into main. The solution is way too heavy to try out a single branch.

**Comment by tagcode** [comment]
Here are some more [workarounds](https://github.com/stride3d/stride/pull/1604).

**Comment by tebjan** [comment]
Does this need an update after this merge? https://github.com/stride3d/stride/pull/1428
